e2e/app/element/base-element.ts

```diff
@@ -252,22 +280,23 @@ export default class BaseElement extends Container {
    * @param text
    */
   async paste(text: string): Promise<void> {
-    const elemt = await this.asElementHandle();
-    return this.page.evaluate((elem, textValue) => {
-      // Refer to https://stackoverflow.com/a/46012210/440432
-      const nativeInputValueSetter = Object.getOwnPropertyDescriptor(window.HTMLTextAreaElement.prototype, 'value').set;
-      nativeInputValueSetter.call(elem, textValue);
-      const event = new Event('input', {bubbles: true});
-      elem.dispatchEvent(event);
-    }, elemt, text);
+    return this.asElementHandle()
+      .then(element => {
+        return this.page.evaluate((elemt, textValue) => {
+          // Refer to https://stackoverflow.com/a/46012210/440432
+          const nativeInputValueSetter = Object.getOwnPropertyDescriptor(window.HTMLTextAreaElement.prototype, 'value').set;
+          nativeInputValueSetter.call(elemt, textValue);
+          const event = new Event('input', { bubbles: true });
+          elemt.dispatchEvent(event);
+        }, element, text);
+      });
   }
 
   /**
    * Returns ElementHandle.
    */
-  async asElementHandle(): Promise<ElementHandle | null> {
-    await this.waitForXPath();
-    return this.element.asElement();
+  async asElementHandle(): Promise<ElementHandle> {
+    return this.waitForXPath();
```

Comment:
We are doing return without await because you need to return Promise rather than value?